shoop/campaigns/models/basket_conditions.py

```diff
@@ -94,3 +94,53 @@ def values(self):
     @values.setter
     def values(self, value):
         self.products = value
+
+
+class ContactGroupBasketCondition(BasketCondition):
+    model = ContactGroup
+    identifier = "basket_contact_group_condition"
+    name = _("Contact Group")
+
+    contact_groups = models.ManyToManyField(ContactGroup, verbose_name=_("contact groups"))
+
+    def matches(self, basket, lines=[]):
+        if basket.customer:
+            contact_groups = basket.customer.groups.all().values_list("pk", flat=True)
+            return self.contact_groups.filter(pk__in=contact_groups).exists()
```

Comment:
There is no need to do `values_list` here.

Doing just:

```python
customers_groups = basket.customer.groups.all()
return self.contact_groups.filter(pk__in=customers_groups).exists()
```

would be faster, since it is only one query.